### PR TITLE
Improve Ci in forks

### PR DESCRIPTION
**Pull request by jsoref**
#### Name of feature:
CI in forks w/ restrictive default permissions

#### Pain or issue this feature alleviates:
#769

#### Supporting links/other PRs/issues:

💔Thank you!

### CODE REVIEW

**Comment by jsoref** [comment]
Testing was performed for the triage workflow here: https://github.com/check-spelling/smallstep-cli/actions/workflows/triage.yml

You can see that it didn't fail for runs 2/3/4. For the final run, I added a dummy secret. That resulted in it *trying* to perform the action, but since the secret wasn't proper, I got a failure, which is sufficient proof that if I had a valid secret it would have worked just as it would work here in a repository which has a secret.

The actions/contents/security-events items appear correct, but the CI fails because I don't have codecov configured which is not a permission (but presumably is a secret). I suppose I could look into that, but if I can leave it as an exercise (you can effectively copy the style from the triage project code...), that'd be great.

File: .github/workflows/triage.yml [hunk 1]
```diff
@@ -4,26 +4,39 @@ on:
...
+        shell: bash
+        if: env.HAS_TRIAGE_PAT == 'true'
+        run: |
+          echo '::set-output name=triage::1'
+
       - uses: actions/add-to-project@v0.3.0
+        if: steps.check.outputs.triage
         with:
           project-url: https://github.com/orgs/smallstep/projects/94
           github-token: ${{ secrets.TRIAGE_PAT }}
```
Comment (dopey) [inline .github/workflows/triage.yml:42]:
```suggestion
  add-to-project:
    name: Add to Triage Project
    runs-on: ubuntu-latest
    env: 
      HAS_TRIAGE_PAT: ${{ secrets.TRIAGE_PAT != '' }}
    if: env.HAS_TRIAGE_PAT == 'true'
    steps:
      - name 
         uses: actions/add-to-project@v0.3.0
         with:
           project-url: https://github.com/orgs/smallstep/projects/94
           github-token: ${{ secrets.TRIAGE_PAT }}
```

**Comment by dopey** [comment]
Codecov isn't a secret. Public repos don't require a codecov token, and shouldn't fail if you provide an empty one. We've got a bunch of public repos that are not providing a CODECOV token, but all of them use that same flow.

**Review by dopey — DISMISSED** [review]

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -8,6 +8,9 @@ on:
 
 jobs:
   ci:
+    permissions:
+      actions: read
```
Comment (dopey) [inline .github/workflows/release.yml:12]:
```suggestion
      actions: read
      contents: read
```

**Review by dopey — APPROVED** [review]

**Pull request merged**